File: testdata/php/level-13-interfaces-implementations/src/animals.php
<?php

interface Animal {
    public function speak(): string;
}

class Dog implements Animal {
    public function speak(): string {
        return "woof";
    }
}

class Cat implements Animal {
    public function speak(): string {
        return "meow";
    }
}

function makeNoise(): void {
    $d = new Dog();
    echo $d->speak();
}

File: testdata/cpp/level-10-virtual-override-resolution/src/animals.cpp
class Animal {
public:
    virtual void speak() = 0;
    virtual int legs() { return 4; }
};

class Dog : public Animal {
public:
    void speak() override {}
    int legs() override { return 4; }
};

class Snake : public Animal {
public:
    void speak() override {}
    int legs() override { return 0; }
};
